src/itemstackmetadata.cpp

```diff
@@ -9,6 +9,22 @@
 #define DESERIALIZE_KV_DELIM_STR "\x02"
 #define DESERIALIZE_PAIR_DELIM_STR "\x03"
 
+#define TOOLCAP_KEY "__tool_capabilities"
```

Comment:
My reasoning for naming it with an "internal"-looking name is that semantically the field isn't "just a string", it's a specifically formatted string and the field is mostly meant as an implementation detail of set_tool_capabilities. The other special fields are meant to be in a format that is directly settable from the metaref, so it makes more sense for the modder to deal with them directly, hence normal-looking names.

I'm not arguing that this is the only way to think about it, this is just a suggestion of a convention for meta fields that aren't meant to be set directly. Maybe it would be useful in another feature to disallow setting meta fields with a key with a special prefix.